### ISSUE TITLE
Completion in namespace

### ISSUE THREAD

**Issue by politza** [Sarcasm/irony-mode#198]
This simple program

    #include <iostream>

    using namespace std;

    int main(int nargs, char **args)
    {
      c|
    }

with `|` marking point, makes irony not offer any completion, unless I prefix the `c` with a `std:`. Am I doing something wrong ?

**Comment by Sarcasm** [comment]
How about `(setq company-minimum-prefix-length 0)`?

Irony will trigger completion automatically after `.`, `->`, `::`, but otherwise the company minimum prefix is necessary.

**Comment by politza** [comment]
Guillaume Papin <notifications@github.com> writes:

> How about (setq company-minimum-prefix-length 0)?

I'm using completion-at-point-functions and irony-completion-at-point.
Debugging tells me that irony-completion-candidates-available-p returns
nil in this scenario.

**Comment by Sarcasm** [comment]
Does it work when using `irony-completion-at-point-async`?

**Comment by politza** [comment]
Guillaume Papin <notifications@github.com> writes:

> Does it work when using irony-completion-at-point-async?

Yes.

**Comment by Sarcasm** [comment]
`completion-at-point` doesn't support asynchronous operation and that is the only operating mode for completion, for now.

**Comment by politza** [comment]
Guillaume Papin <notifications@github.com> writes:

> completion-at-point doesn't support asynchronous operation and that is
> the only operating mode for completion, for now.

But enabling irony-mode puts irony-completion-at-point, via
irony-completion--enter, on the completion-at-point-functions hook ...

**Comment by Sarcasm** [comment]
Yeah I know, that's why the configuration section recommend to remap completion at point with irony's async version :https://github.com/Sarcasm/irony-mode/blob/master/README.md#configuration

Hopefully in the future irony will have a synchronous version and completion-at-point will provide support for asynchronous completion-function

**Comment by politza** [comment]
Ah, OK.
